gph/src/ripser.h

```diff
@@ -691,9 +691,32 @@ class ripser
         };
 
         const index_t cnt = n - (k - 1);
+
         if (pred(n) || (cnt <= 0))
             return n;
-        return get_max(n, cnt, pred);
+
+        if (k != 3) {
+            return get_max(n, cnt, pred);
+        } else {
+            long double to_cbrt = 6 * idx;
+            index_t guess = static_cast<index_t>(std::floor(std::cbrt(to_cbrt)));
+            if (!pred(guess)) {
+                while (true) {
+                    --guess;
+                    if (pred(guess))
+                        break;
+                }
+            } else {
+                while (true) {
+                    ++guess;
+                    if (!pred(guess)) {
+                        --guess;
+                        break;
+                    }
+                }
+            }
```

Comment:
This routine is based on the observation that `guess` seems to always be within distance 1 of the true value.  Likely there is a closed-form formula with some tiny correction that we could work out, but I think we'd still need the left-and-right checks due to numerics that become more important for 2D simplices given the large index values.

Reply:
Do you have a link about the theory of this or is this common in your field ?

Reply:
The cube root is "obvious", and even the leading 6 can be understood easily I think.  But for now I just have verification until very, very large numbers (I need to get my files again to tell you quite how large).

Reply:
In any case, this solution should always be correct as long as the initial guess is tested.